### PR TITLE
Add EIP: Financial Bonds

### PR DESCRIPTION
**Pull request by Edoumou**
When opening a pull request to submit a new EIP, please use the suggested template: https://github.com/ethereum/EIPs/blob/master/eip-template.md

We have a GitHub bot that automatically merges some PRs. It will merge yours immediately if certain criteria are met:

 - The PR edits only existing draft PRs.
 - The build passes.
 - Your GitHub username or email address is listed in the 'author' header of all affected PRs, inside <triangular brackets>.
 - If matching on email address, the email address is the one publicly listed on your GitHub profile.

### CODE REVIEW

File: EIPS/eip-7092.md [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+* @title ERC-7092 Financial Bonds tandard
+*/
+interface IERC7092 {
+    /**
+    *  @notice Returns the bond isin
+    */
+    function isin() external view returns(string memory);
+
+    /**
+    * @notice Returs the bond name
```
Comment (fosgate29) [inline EIPS/eip-7092.md:49]: typo "Returns"

File: EIPS/eip-7092.md [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+    */
+    function name() external view returns(string memory);
+
+    /**
+    * @notice Returns the bond symbol
+    */
+    function symbol() external view returns(string memory);
+
+    /**
+    * @notice Returns the numbr of decimals the bond uses - e?g `10`, means to divide the token amount by `10000000000`
```
Comment (fosgate29) [inline EIPS/eip-7092.md:59]: typo: "number of decimals"

File: EIPS/eip-7092.md [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+    */
+    function currency() external view returns(address);
+
+    /**
+    * @notice Returns the copoun currency. This is the contract address of the token used to pay coupons. It can be same as the the one used for the principal
+    */
+    function currencyOfCoupon() external view returns(address);
+
+    /**
+    * @notice Returns the bond denominiation. This is the minimum amount in which the Bonds may be issued. It must be expressend in unnit of the principal currency
```
Comment (fosgate29) [inline EIPS/eip-7092.md:76]: typo: instead of  "unnit" it should be "uint"

File: assets/eip-7092/IERC7092.sol [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+* @title ERC-7092 Financial Bonds tandard
+*/
+interface IERC7092 {
+    /**
+    *  @notice Returns the bond isin
+    */
+    function isin() external view returns(string memory);
+
+    /**
+    * @notice Returs the bond name
```
Comment (fosgate29) [inline assets/eip-7092/IERC7092.sol:14]: typo: "Returns"

**Review by fosgate29 — COMMENTED** [review]
Just fixed a couple of typos

File: EIPS/eip-7092.md (hunk 3 above)
Comment (Edoumou) [reply EIPS/eip-7092.md:76]: Yes @fosgate29 it should.

Thx

File: assets/eip-7092/IERC7092.sol [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+    */
+    function name() external view returns(string memory);
+
+    /**
+    * @notice Returns the bond symbol
+    */
+    function symbol() external view returns(string memory);
+
+    /**
+    * @notice Returns the numbr of decimals the bond uses - e?g `10`, means to divide the token amount by `10000000000`
```
[the remark this answers is not in the document]
Comment (Edoumou) [reply assets/eip-7092/IERC7092.sol:24]: Thank you,

Going to fix all these typos

[the remark this answers is not in the document]
Comment (Edoumou) [reply assets/eip-7092/IERC7092.sol:24]: Going to fix them all.

🙏🙏🙏

**Comment by Edoumou** [comment]
Hello @axic, @Pandapip1, @SamWilsn, @xinbenlv

I submitted this ERC for financial bonds since May 29, all checks have passed but there is no review since the submission.

Is there something missing in order for the PR to be merged ?

File: EIPS/eip-7092.md [hunk 6]
```diff
@@ -0,0 +1,342 @@
+---
+eip: 7092
+title: Financial Bonds
+description: This interface defines a specification for financial bonds tokenization
```
Comment (SamWilsn) [inline EIPS/eip-7092.md:4]: This description restates the title too much. Try to add new information in the description that isn't in the title.

File: EIPS/eip-7092.md [hunk 7]
```diff
@@ -0,0 +1,342 @@
...
+created: 2023-05-28
+---
+
+## Abstract
+
+The proposed standard allows for the implementation of basic functionality for fixed income financial bonds with smart contracts.
+Principal bonds characteristics such as the bond isin, the issue volume, the issue date, the maturity date, the coupon rate,
+the coupon frequency, the principal, or the day count basis are defined to allow issuing bonds in the primary market (origination),
+and different transfer functions allow to buy or sell bonds in the secondary market. The standard also providses a functionality to
+allow bonds to be approved by owners in order to be spent by third party.
```
Comment (SamWilsn) [inline EIPS/eip-7092.md:19]:
```suggestion
This proposal introduces fixed income financial bonds. Important bonds characteristics such as the International Securities Identification Number (ISIN), the issue volume, the issue date, the maturity date, the coupon rate,
the coupon frequency, the principal, or the day count basis are defined to allow issuing bonds in the primary market (origination), and different transfer functions allow to buy or sell bonds in the secondary market. The standard also provides a functionality to allow bonds to be approved by owners in order to be spent by third party.
```

File: EIPS/eip-7092.md [hunk 8]
```diff
@@ -0,0 +1,342 @@
...
+
+## Motivation
+
+Fixed income instruments are one of the asset classes that is widely used by corporations and other entities to raise funds. Bonds
+are considered more secured than equity since the issuer is supposed to repay the principal at maturity in addition to coupons
+that are paid to investsors.
+
+This standard interface allows fixed income instruments to be represented as on-chain tokens, so as they can be managed through wallets,
+and be used by applications like decentrailized exchanges.
+
```
Comment (SamWilsn) [inline EIPS/eip-7092.md:29]: Are there any existing standards for bonds on-chain? If so, you should note why this standard is better.

File: EIPS/eip-7092.md [hunk 9]
```diff
@@ -0,0 +1,342 @@
...
+tokenized bonds by integrating this financial bond interface.
+
+Tokenizing bonds will offer several advantages compared to traditional bond issuance and trading, among with:
+
+1. Fractional ownership: The bond standard does not limit the bond denomination to some minimum value compared to traditioanal bonds where the denomination is typically equal to $100 or $1,000.
+2. Accessibility: By allowing lower investment thresholds, tokenized bonds are supposed to attract retail investors who could not participate in traditional markets due to high minimum investment requirements.
+3. Increased liquidity: Fractioanal ownership will bring new investors in the bond market, this will increase liquidity in the bond market.
+4. Cost savings: By replacing intermediaries with smart contracts, bond's tokenization will reduce costs associated with the bond issuance and management.
+5. Easy accessibility and 24/7 trading: Tokenized bonds are supposed to be traded on digital platforms such as decentralized exchanges. Therefore, they will be more accessible compared to tradional bond market.
```
Comment (SamWilsn) [inline EIPS/eip-7092.md:241]: This list belongs in motivation. The rationale section should describe technical choices made within the EIP itself, while the motivation should justify the existence of the EIP as a whole.

A good example of what you _should_ include in your rationale section is why you don't extend an existing token standard, like ERC-20 or ERC-1155.

File: EIPS/eip-7092.md [hunk 10]
```diff
@@ -0,0 +1,342 @@
...
+
+Some bonds have embedded options attached to them. As an example we can cite:
+
+1. Callable bonds that have an option that gives the issuer the right to retire bonds before they mature.
+2. Puttable bonds that have an option that gives investors the right to retire bonds before they mature.
+3. Convertible bonds that gives investors the right to convert their bonds to equity.
+
+Bonds with embedded options can be created by inheriting from the basic ERC-7092 that integrates the proposed interface.
+
+CALLABLE BONDS:
```
Comment (SamWilsn) [inline EIPS/eip-7092.md:255]: You can use a third level heading here (`### Callable Bonds`)

**Review by Frenchy9669 — APPROVED** [review]

**Review by Frenchy9669 — APPROVED** [review]

File: EIPS/eip-7092.md [hunk 11]
```diff
@@ -0,0 +1,412 @@
+---
+eip: 7092
+title: Financial Bonds
+description: This interface defines a specification for debt issued by corporations, Goverments or other entities to investors in order to raise funds.
```
Comment (lightclient) [inline EIPS/eip-7092.md:4]:
```suggestion
description: This interface defines a specification for debt issued by corporations, governments, or other entities to investors in order to raise funds.
```

File: EIPS/eip-7092.md [hunk 12]
```diff
@@ -0,0 +1,412 @@
...
+discussions-to: https://ethereum-magicians.org/t/financial-bonds/14461
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-05-28
+---
+
+## Abstract
+
+This proposal introduces fixed income financial bonds. Important bonds characteristics such as the International Securities Identification Number (ISIN),
```
Comment (lightclient) [inline EIPS/eip-7092.md:15]:
```suggestion
This proposal introduces fixed income financial bonds. Important bond characteristics such as the International Securities Identification Number (ISIN),
```
singular?

File: EIPS/eip-7092.md [hunk 13]
```diff
@@ -0,0 +1,412 @@
...
+the issue volume, the issue date, the maturity date, the coupon rate, the coupon frequency, the principal, or the day count basis are defined to allow issuing
+bonds in the primary market (origination), and different transfer functions allow to buy or sell bonds in the secondary market. The standard also provides a
+functionality to allow bonds to be approved by owners in order to be spent by third party.
+
+## Motivation
+
+Fixed income instruments are one of the asset classes that is widely used by corporations and other entities to raise funds. Bonds
```
Comment (lightclient) [inline EIPS/eip-7092.md:22]:
```suggestion
Fixed income instruments is one of the asset classes that is widely used by corporations and other entities to raise funds. Bonds
```

File: EIPS/eip-7092.md [hunk 14]
```diff
@@ -0,0 +1,412 @@
...
+* @title ERC-7092 Financial Bonds tandard
+*/
+interface IERC7092 {
+    /**
+    *  @notice Returns the bond isin
+    */
+    function isin() external view returns(string memory);
+
+    /**
+    * @notice Returs the bond name
```
Comment (lightclient) [inline EIPS/eip-7092.md:65]:
```suggestion
    * @notice Returns the bond name
```

File: EIPS/eip-7092.md [hunk 15]
```diff
@@ -0,0 +1,412 @@
...
+    * @notice Returns the bond symbol
+    *         It is RECOMMENDED to represent the symbol as a combination of the issuer Issuer'shorter name and the maturity date
+    *         Ex: If a company named Green Energy issues bonds that will mature on october 25, 2030, the bond symbol could be `GE30` or `GE2030` or `GE102530`
+    *
+    * OPTIONAL - interfaces and other contracts MUST NOT expect these values to be present. The method is used to improve usability.
+    */
+    function symbol() external view returns(string memory);
+
+    /**
+    * @notice Returns the numbr of decimals the bond uses - e?g `10`, means to divide the token amount by `10000000000`
```
Comment (lightclient) [inline EIPS/eip-7092.md:79]:
```suggestion
    * @notice Returns the number of decimals the bond uses - e?g `10`, means to divide the token amount by `10000000000`
```

File: EIPS/eip-7092.md [hunk 16]
```diff
@@ -0,0 +1,412 @@
...
+
+Because the balance of an account can be derived from both the `principal` and the `denomination`, we made the choice not to define the `balanceOf` function.
+However it is RECOMMENDED to define the `balanceOf` function in any contract that implements this standard. In that case,
+the `balance` of an account MUST be equal to the ratio of the `principal` of that account and the `denomination`.
+
+```javascript
+    balanceOf(account) = principalOf(account) / denomination
+```
+
+### Backwards compatibility
```
Comment (lightclient) [inline EIPS/eip-7092.md:303]:
```suggestion
## Backwards compatibility
```

**Review by lightclient — CHANGES_REQUESTED** [review]

File: EIPS/eip-7092.md (hunk 16 above)
Comment (Edoumou) [reply EIPS/eip-7092.md]: ## Backwards compatibility

doesn't pass the EIP Validator test. This is probably due to the fact that this is not defined in the template and should then be defined as a sub-sub-section.

For this reason, I will keep it as defined previously, i.e ## Backwards compatibility for the test to pass.

Comment (lightclient) [reply EIPS/eip-7092.md:303]: Ah you need to capitalize "Compatibility". Using `###` is incorrect.

File: EIPS/eip-7092.md [hunk 17]
```diff
@@ -300,7 +300,7 @@ the `balance` of an account MUST be equal to the ratio of the `principal` of tha
     balanceOf(account) = principalOf(account) / denomination
 ```
 
-### Backwards compatibility
+## Backwards compatibility
```
Comment (lightclient) [inline EIPS/eip-7092.md:303]: must be `## Backwards Compatibility`

Comment (Edoumou) [reply EIPS/eip-7092.md]: Hello @lightclient 

When I use `## Backwards Compatibility` the test fails. The error message is 'An extra section has been added'. The backward compatibility doesn't appear in the template, that's probably the reason why when we add it as a section with ## the test fails

Comment (Edoumou) [reply EIPS/eip-7092.md]: Yes, I think there was a typo in the previous one. the previous one was `Backwards compatibility`, but it should be `Backwards Compatibility` with capital `C`.

I just corrected it.

**Comment by Edoumou** [comment]
Hello @lightclient

I just add changes you requested. Can you review and approve them ?

File: assets/eip-7092/IERC7092.sol [hunk 18]
```diff
@@ -0,0 +1,186 @@
+// SPDX-License-Identifier: MIT
```
Comment (lightclient) [inline assets/eip-7092/IERC7092.sol:1]:
```suggestion
// SPDX-License-Identifier: CC0-1.0
```

File: assets/eip-7092/ERC7092.sol [hunk 19]
```diff
@@ -0,0 +1,298 @@
+// SPDX-License-Identifier: MIT
```
Comment (lightclient) [inline assets/eip-7092/ERC7092.sol:1]:
```suggestion
// SPDX-License-Identifier: CC0-1.0
```

File: assets/eip-7092/BondStorage.sol [hunk 20]
```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: MIT
```
Comment (lightclient) [inline assets/eip-7092/BondStorage.sol:1]:
```suggestion
// SPDX-License-Identifier: CC0-1.0
```

**Review by lightclient — COMMENTED** [review]
One last issue. All EIPs must be in the public domain.

**Comment by Edoumou** [comment]
> One last issue. All EIPs must be in the public domain.

Hello @lightclient

I just updated the the License from `MIT` to `CC0-1.0`

**Review by lightclient — APPROVED** [review]
Looks good, thanks!

**Pull request merged**